osg-token-renewer.py

```diff
@@ -38,46 +41,63 @@ def validate_config_dict(cfgx):
     for token in cfgx["token"]:
         account = cfgx["token"][token].get("account")
         if not account:
-            print("token %s: missing 'account' attribute" % token,
-                  file=sys.stderr)
-            return False
+            return emsg(f"token {token}: missing 'account' attribute")
         elif account not in cfgx["account"]:
-            print("token %s: missing 'account %s' section" % (token, account),
-                  file=sys.stderr)
-            return False
+            return emsg(f"token {token}: missing 'account {account}' section")
         elif not cfgx["account"][account].get("password_file"):
-            print("account %s: missing 'password_file' attribute" % account,
-                  file=sys.stderr)
-            return False
+            return emsg(f"account {account}: missing 'password_file' attribute")
+        elif not cfgx["token"][token].get("token_path"):
+            return emsg(f"token {token}: missing 'token_path' attribute")
 
     return True
 
 
 def add_all_accounts(cfgx):
     accounts = cfgx["account"]
     added = set()
+    errors = 0
 
     for token in cfgx["token"]:
         acct = cfgx["token"][token].get("account")
         if acct in added:
             continue
         print("account %s" % acct)
         pwfile = accounts[acct]['password_file']
-        add_account(acct, pwfile)
+        try:
+            add_account(acct, pwfile)
+        except subprocess.CalledProcessError as e:
+            emsg(f"Failed to create account '{acct}': {e}")
+            errors += 1
         added.add(acct)
 
+    return errors
+
 
 def make_all_tokens(cfgx):
     tokens = cfgx["token"]
+    errors = 0
 
     for t in tokens:
         print("token %s" % t)
         for k,v in tokens[t].items():
             print("{}: {}".format(k,v))
         print("---")
-        mktoken(tokens[t])
+        try:
+            mktoken(tokens[t])
+        except subprocess.CalledProcessError as e:
+            emsg(f"Failed to create token '{t}': {e}")
+            errors += 1
+        except subprocess.CalledProcessError as e:
+            emsg(f"Failed to create token '{t}': {e}")
+            errors += 1
```

Comment:
Any reason for this duplication?

Reply:
Just to keep you on your toes?  ;)

No it looks like an accidental extra copy - removing!